### ISSUE TITLE
Python tiledbsoma SOMACollection.__repr__ is very slow

### ISSUE THREAD

**Issue by bkmartinjr** [single-cell-data/TileDB-SOMA#301]
I've been doing a bit of notebook/ipytyhon programming, and hitting the object __repr__ a lot.  If the objects are stored in S3, it is very slow when compared to the native TileDB repr.  Example, using the two API on the same URI.
* tiledb - 1.4 sec
* soma - 52 sec

I don't think they need to be identical, but a minute is too long for what should be an interactive response.

```
In [26]: %time print(repr(tiledb.Group("s3://czi.bruce-public/tmp/atlas-20220916/")))
 GROUP
|-- human GROUP
|------ ms GROUP
|---------- RNA GROUP
|-------------- var ARRAY
|-------------- X GROUP
|------------------ raw ARRAY
|------------------ final ARRAY
|------ obs ARRAY
|-- mouse GROUP
|------ ms GROUP
|---------- RNA GROUP
|-------------- var ARRAY
|-------------- X GROUP
|------------------ raw ARRAY
|------------------ final ARRAY
|------ obs ARRAY

CPU times: user 98.7 ms, sys: 29.7 ms, total: 128 ms
Wall time: 1.38 s

In [27]: %time print(repr(soma.SOMACollection("s3://czi.bruce-public/tmp/atlas-20220916/")))
 SOMACollection:
  human SOMAExperiment:
    ms SOMACollection:
      RNA SOMAMeasurement:
        var SOMADataFrame (18446744073709551615,)
        X SOMACollection:
          raw SOMASparseNdArray (21542706, 60567)
          final SOMASparseNdArray (21542706, 60567)
    obs SOMADataFrame (18446744073709551615,)
  mouse SOMAExperiment:
    ms SOMACollection:
      RNA SOMAMeasurement:
        var SOMADataFrame (18446744073709551615,)
        X SOMACollection:
          raw SOMASparseNdArray (3975444, 62628)
          final SOMASparseNdArray (3975444, 62628)
    obs SOMADataFrame (18446744073709551615,)
CPU times: user 7.31 s, sys: 2.93 s, total: 10.2 s
Wall time: 52.3 s
```

**Comment by johnkerl** [comment]
Analysis is on #303.

**Comment by johnkerl** [comment]
cc @ihnorton @Shelnutt2 re perf of `tiledb.open` at this scale even on in-region, not-over-the-internet access

**Comment by bkmartinjr** [comment]
I have not been able to get consolidation to work due to the unreleased TileDB-Inc/TileDB#3518

As soon as that fix is released, I will re-generate the dataset and try this test.

**Comment by ihnorton** [comment]
I'll do a release tomorrow with that fix, we need to ship another fix along with this one (fix has active PR, waiting on CI and final tests).

**Comment by bkmartinjr** [comment]
This appears to be resolved, except for some follow-up captured in #392.  Closing.
